### PR TITLE
Add VGG16 with basic image preprocessing

### PR DESCRIPTION
**Pull request by oarriaga**

### CODE REVIEW

**Comment by paganpasta** [comment]
Can we make the implementation general for the VGG family? 
What I mean is VGG16 is quite a specific model and others like VGG11(+BN), VGG19(+BN) etc can also be supported with minimal effort.

File: eqxvision/models/classification/vgg16.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import jax
+import jax.numpy as jnp
+from equinox import Module
+from equinox.nn import MaxPool2D, Linear, Conv2d
+
+
+def preprocess_input(image, mean=[103.939, 116.779, 123.68]):
```
Comment (paganpasta) [inline eqxvision/models/classification/vgg16.py:7]: This is usually taken care at the data loading side of things as people tend to apply different transformations as per their needs.

**Comment by paganpasta** [comment]
Just a reminder for later stages, missing test cases to verify that forward pass is going through with and without jit.

File: eqxvision/models/classification/vgg16.py (hunk 1 above)
Comment (oarriaga) [reply eqxvision/models/classification/vgg16.py:7]: If one uses the pre-trained weights and wants to use the model directly, then one would need this specific transformation.

**Comment by oarriaga** [comment]
Unfortunately, at the moment I don't have much time to think about how one could make it general for other VGG's. But after this PR someone with more time can take over that task. Moreover, I think that independent code bases between VGG16 or VGG19 might be better, rather than abstracting some small generalizations. For example [VGG16](https://github.com/keras-team/keras/blob/v2.9.0/keras/applications/vgg16.py#L43) and [VGG19](https://github.com/keras-team/keras/blob/v2.9.0/keras/applications/vgg19.py#L43) model implementations in Keras are independent of each other, and this makes it easy and simple to read and understand. What do you think?

**Comment by paganpasta** [comment]
The issue I raised this is because existing model architectures in eqxvision support model_* variations and just supporting VGG16 for VGG will break the consistency for the rest. Plus, having multiple classes declared (with/wo BatchNorm) also increases duplicacy of the code. 
Don't worry about the time I can take it over from here for this else if you still wish to push for it then you are more than welcome!

**Comment by paganpasta** [comment]
@oarriaga Sorry, just wanted to confirm, should I close the merge? I can bundle in VGG with test case revamp I was doing.

**Comment by oarriaga** [comment]
Ah OK I was thinking that you could accept the PR and continue developing with another PR or just commits to master.

**Comment by paganpasta** [comment]
Sorry for the confusion.
